src/Evolution/Systems/ScalarWave/BoundaryConditions/ConstraintPreservingSphericalRadiation.cpp

```diff
@@ -0,0 +1,178 @@
+  const auto type_read = options.parse_as<std::string>();
+  if ("Sommerfeld" == type_read) {
+    return ConstraintPreservingSphericalRadiationType::Sommerfeld;
+  } else if ("FirstOrderBaylissTurkel" == type_read) {
+    return ConstraintPreservingSphericalRadiationType::FirstOrderBaylissTurkel;
+  } else if ("SecondOrderBaylissTurkel" == type_read) {
+    return ConstraintPreservingSphericalRadiationType::SecondOrderBaylissTurkel;
+  }
+  PARSE_ERROR(options.context(),
+              "Failed to convert \""
+                  << type_read
+                  << "\" to ConstraintPreservingSphericalRadiation::Type. Must "
+                     "be one of Sommerfeld, FirstOrderBaylissTurkel, or "
+                     "SecondOrderBaylissTurkel.");
+}
+}  // namespace detail
+
+template <size_t Dim>
+ConstraintPreservingSphericalRadiation<Dim>::
+    ConstraintPreservingSphericalRadiation(
+        const detail::ConstraintPreservingSphericalRadiationType type) noexcept
+    : type_(type) {}
+
+template <size_t Dim>
+ConstraintPreservingSphericalRadiation<Dim>::
+    ConstraintPreservingSphericalRadiation(CkMigrateMessage* const msg) noexcept
+    : BoundaryCondition<Dim>(msg) {}
+
+template <size_t Dim>
+std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>
+ConstraintPreservingSphericalRadiation<Dim>::get_clone() const noexcept {
+  return std::make_unique<ConstraintPreservingSphericalRadiation>(*this);
+}
+
+template <size_t Dim>
+void ConstraintPreservingSphericalRadiation<Dim>::pup(PUP::er& p) {
+  BoundaryCondition<Dim>::pup(p);
+  p | type_;
+}
+
+template <size_t Dim>
+std::optional<std::string>
+ConstraintPreservingSphericalRadiation<Dim>::dg_time_derivative(
+    const gsl::not_null<Scalar<DataVector>*> dt_pi_correction,
+    const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+        dt_phi_correction,
+    const gsl::not_null<Scalar<DataVector>*> dt_psi_correction,
+    const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+        face_mesh_velocity,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_covector,
+    const Scalar<DataVector>& pi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& phi,
+    const Scalar<DataVector>& psi,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& coords,
+    const Scalar<DataVector>& gamma2, const Scalar<DataVector>& dt_pi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& dt_phi,
+    const Scalar<DataVector>& dt_psi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& d_pi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& d_psi,
+    const tnsr::ij<DataVector, Dim, Frame::Inertial>& d_phi) const noexcept {
+  {
+    DataVector& inv_radius = get(*dt_psi_correction);
+    inv_radius = square(get<0>(coords));
+    for (size_t i = 1; i < Dim; ++i) {
+      inv_radius += square(coords.get(i));
+    }
+    inv_radius = 1.0 / sqrt(inv_radius);
+    if (type_ ==
+        detail::ConstraintPreservingSphericalRadiationType::Sommerfeld) {
```

Comment:
use `switch`?

Reply:
I find multi-line switch statements a bit annoying and difficult to read